lib/administration/user/include/irods/user_administration.hpp

```diff
@@ -74,7 +74,19 @@ namespace irods::experimental::administration
     /// \since 4.3.1
     struct user_password_property
     {
-        std::string value;
+        explicit user_password_property(std::string _value, std::optional<std::string> _key = std::nullopt)
+            : value{std::move(_value)}
+            , key{std::move(_key)}
+        {
+        }
+
+        // NOLINTBEGIN(misc-non-private-member-variables-in-classes)
+        std::string value; ///< The new password.
+
+        /// The password of the user requesting the change.
+        /// If not set, obfGetPw() is used. Use of obfGetPw() implies a .irodsA file exists.
+        std::optional<std::string> key;
```

Comment:
`key` and `value` are somewhat related.

I chose `key` because I couldn't come up with anything else at the time. It is the caller's deobfuscated password. It exists for cases where calling `obfGetPw` is impossible and/or not preferred.

I didn't want to put what it actually was, but the comment implies what is happening for anyone familiar with the code.

Alternative naming welcome.

Reply:
Some ideas...
 - `requesting_user_password`
 - `credentials`
 - `caller_credentials`
 - `requesting_user_credentials`

I just feel like `key` forms a relationship to `value` which doesn't necessarily exist. It's not a big deal, it just confused me when I saw it.

Reply:
Totally understand. Will change.

Reply:
Changed to `requester_password`.

I considered changing it to `requesting_user_password`, but that's a lot of words lol and `requester` implies `user` to an extent.

Reply:
I like it

Reply:
Cool cool.